lib/maps.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+var Stream = require('stream')
+  , util = require('util')
+  , request = require('request')
+  , debug = require('debug')('cartodb')
+  , resources = require('./resources.json')
+  , Mustache = require('mustache')
+  , _Promise = require('./promise')
+  ;
+
+/*
+ * Inheritance
+ */
+
+util.inherits(Named, Stream);
+
+
+function Named(config) {
+
+  for(var i in config){
+    this[i] = config[i];
+  }
+
+  this.maps_api_url = this.maps_api_url || Mustache.render(resources.maps_api_url, config);
+
+  return this;
+}
+
+Named.prototype.create = function(options) {
+  debug('POSTing your template to %s', this.maps_api_url);
+
+  var self = this;
+  var promise = new _Promise();
+
+  request.post({
```

Comment:
How about factoring all the request calls in some utility method?
```
Named.prototype._getRequestPromise = function(method, body, url, callback) {
  return request[method]({
    url: url,
    body: body,
    qs: ...
  }, callback
}
```
could even be on a separate utils module.